hera_cal/frf.py

```diff
@@ -0,0 +1,354 @@
+
+    def timeavg_data(self, t_avg, rephase=False, verbose=True):
+        """
+        Time average data attached to object given a averaging time-scale t_avg [seconds].
+        The time-averaged data, flags, time arrays, etc. are stored in avg_* attributes.
+        Note that although denoted avg_flags for consistency, this array stores the AND 
+        of flags in each averaging window.
+
+        The t_avg provided will be rounded to the nearest time that makes Navg 
+        an integer, and is stored as self.t_avg.
+
+        Parameters
+        ----------
+        t_avg : float
+            Width of time-averaging window in seconds.
+        """
+        # turn t_avg into Navg given dtime
+        Navg = int(np.round((t_avg /  (3600.0 * 24) / self.dtime)))
+        assert Navg > 0, "A t_avg of {:0.5f} makes Navg=0, which is too small.".format(t_avg)
+        if Navg > self.Ntimes:
+            Navg = self.Ntimes
+        old_t_avg = t_avg
+        t_avg = Navg * self.dtime * 3600.0 * 24
+
+        if verbose: 
+            print "The t_avg provided of {:.1f} has been shifted to {:.1f} to make Navg = {:d}".format(old_t_avg, t_avg, Navg)
+
+        # setup lists
+        avg_data = odict()
+        avg_flags = odict()
+        avg_nsamples = odict()
+
+        # iterate over keys
+        for i, k in enumerate(self.data.keys()):
+            ad, af, an, al, ea = timeavg_waterfall(self.data[k], Navg, flags=self.flags[k], nsamples=self.nsamples[k],
+                                                   rephase=rephase, lsts=self.lsts, freqs=self.freqs, bl_vec=self.blvecs[k[:2]],
+                                                   lat=self.lat, extra_arrays=dict(times=self.times), verbose=verbose)
+            avg_data[k] = ad
+            avg_flags[k] = af
+            avg_nsamples[k] = an
+
+        self.avg_data = datacontainer.DataContainer(avg_data)
+        self.avg_flags = datacontainer.DataContainer(avg_flags)
+        self.avg_nsamples = datacontainer.DataContainer(avg_nsamples)
+        self.avg_lsts = al
+        self.avg_times = ea['avg_times']
+        self.t_avg = t_avg
+        self.Navg = Navg
+
+    def write_data(self, outfilename, write_avg=True, filetype='miriad', add_to_history='', overwrite=False):
+        """
+        Write data in FRFringe object. If write_avg == True, write the self.avg_data dictionary,
+        else write the self.data dictionary.
+
+        Parameters
+        ----------
+        outfilename : str
+            Path to output visibility data.
+
+        write_avg : bool
+            If True, write the avg_data dictionary, else write the data dictionary.
+
+        filetype : str
+            Output file format. Currently only miriad is supported.
+
+        add_to_history = str
+            History string to add to existing UVData object attached as self.inp_uvdata.
+
+        overwrite: bool
+            If True, overwrite output if it exists.
+        """
```

Comment:
You're missing documentation on your return statement.

Reply:
thanks